### ISSUE TITLE
How to send ajax POST url and POST data?

### ISSUE THREAD

**Issue by MatthewPattell** [saribe/eModal#36]
I want send post request to url and post data. How to do it?
And how to set modal container attribute "id" or "class"?

**Comment by saribe** [comment]
can you rephrase this: "I want send post request to url and post data. How to do it?"

About id or class, roght now it's not possible, however, when you do eModal.####();

this will return a promise, but, in promise object you can find "element", this is the DOM (jquery dom object) for Modal.


Please feel free to contact me if you need any further information.

**Comment by MatthewPattell** [comment]
About id or class: Ok, I clear.

I paraphrasing:
Let's say I need to send data using the POST method. How can i do this?
(example: eModal.ajax({url:"/test/", data:{id:1,test:4}}))

**Comment by angelzeke** [comment]
l 31
var datas = {};
l 297
        function ajax(data, title) {
            var dfd = _createDeferred();

            var params = {
                async: true,
                deferred: dfd,
                loading: true,
                title: data.title || title || defaultSettings.title,
                url: data.url || data,
                dataType: 'text'
            };

            if(data.datas){
                params.method="POST";
                params.data=data.datas;
            }

Example: eModal.ajax({url:"/test/", datas:{id:1,test:4}}) <-- $.ajax({method: "POST"});

**Comment by saribe** [comment]
All params send to eModal.ajax will be set on jQuery request, so if you set eModal.ajax({method: "POST", ...}) it will be a POST.

check if you have >= V 1.2.61

**Comment by saribe** [comment]
Now it is possible to set id on eModal

`eModal.setId('your-id');`
